app/src/main/resources/static/jaewon/signup/signup.js

```diff
@@ -0,0 +1,236 @@
+			if (!target.value) {
+				isValid = false;
+			}
+		});
+		if (isValid === false) {
+			window.alert("필수 항목을 입력하세요")
+			return false;
+		}
+		var fd = new FormData(document.forms.namedItem("signup"));
+		fd.append('type', '1');
+		xLocallist = document.querySelector("#locallist")
+		for(i=0; i<xLocallist.children.length; i++){
+			var value = xLocallist.children[i].value
+			fd.append('localsArr', value);
+		}
+		xPupslist = document.querySelector("#pupslist")
+		for(i=0; i<xPupslist.children.length; i++){
+			var value = xPupslist.children[i].value
+			fd.append('pupsArr', value);
+		}
+		
+
+
+		fetch("/member/signup", {
+			method: "POST",
+			body: new URLSearchParams(fd)
+		}).then(function(response) {
+			return response.json();
+		}).then(function(result) {
+			if (result.status == "success") {
+				location.href = "/junho/index.html";
+			} else {
+				window.alert("로그인 실패!")
+			}
+		});
+		return false;
+	});
+}
+
+
+
+function locallistsi() {
+	var locallistsi = document.querySelector("#localboard");
+	var Selectsi = document.querySelector("#select-si");
+	var htmlGenerators = Handlebars.compile(Selectsi.innerHTML);
+	fetch("/activeLocal/list")
+		.then(function(response) {
+			return response.json();
+		}).then(function(result) {
+			if (result.status == "fail") {
+				window.alert("서버 요청 오류!");
+				console.log(result.data);
+				return;
+			}
+			locallistsi.innerHTML = htmlGenerators(result);
+		});
+}
+
+function locallistgu() {
+	var locallistgu = document.querySelector("#locallistgu");
+	var Selectgu = document.querySelector("#select-gu");
+	var htmlGeneratorg = Handlebars.compile(Selectgu.innerHTML);
+	$('#localboard').change(function() {
+		var gu = $("#localboard option:selected").val()
+		var qs = `nameGu=${gu}`
+		fetch(`/activeLocal/list-gu?${qs}`)
+			.then(function(response) {
+				return response.json();
+			}).then(function(result) {
+				if (result.status == "fail") {
+					window.alert("서버 요청 오류!");
+					console.log(result.data);
+					return;
+				}
+				locallistgu.innerHTML = htmlGeneratorg(result);
+			});
+	})
+}
+
+function addlocal() {
+	var locallist = document.querySelector("#locallist");
+	var selectlocal = document.querySelector("#selectlocal");
+	var htmlGeneratorselect = Handlebars.compile(selectlocal.innerHTML);
+	$('.lo-btn').on('click', (e) => {
+		var gu = $("#locallistgu option:selected").val()
+		if (gu === '') {
+			alert("시/구를 입력해주세요");
+			return false
+		}
+		var qs = `no=${gu}`
+		fetch(`/activeLocal/get?${qs}`)
+			.then(function(response) {
+				return response.json();
+			}).then(function(result) {
+				if (result.status == "fail") {
+					window.alert("서버 요청 오류!");
+					console.log(result.data);
+					return;
+				}
+
+				for (i = 0; i < $(locallist).children().length; i++) {
+					if ($(locallist).children()[i].value == result.no) {
+						console.log(result.no)
+						alert("중복된 지역은 추가할 수 없습니다.")
+						return false;
+					}
+				}
+
+				if ($(locallist).children().length > 2) {
```

Comment:
```suggestion
var $localListChildren = $(locallist).children();

for (i = 0; i < $localListChildren.length; i++) {

if ($localListChildren.eq(i).val() == result.no) {

if ($localListChildren.length > 2) {
```
- $(locallist).children() 를 3군데서 사용하고 있음
- 해당 구문마다 새로 셀레트하는 거라서
- 위에 변수로 선언하고 사용하는 것으로 변경